webapp/src/js/enketo/medic-xpath-extensions.js

```diff
@@ -56,10 +64,51 @@ const parseTimestampToDate = (value) => {
   const timestamp = parseInt(getValue(value));
 
   if (isNaN(timestamp)) {
-    return { t: 'str', v: '' };
+    return XPR.string('');
   }
 
-  return { t:'date', v: new Date(timestamp) };
+  return XPR.date(new Date(timestamp));
+};
+
+const asString = (r) => {
+  return r.t === 'arr' ?
+    r.v.length && !isNil(r.v[0]) ? r.v[0].textContent || '' : '' :
+    r.v.toString();
+};
+
+// Copied from https://github.com/enketo/openrosa-xpath-evaluator/blob/master/src/openrosa-extensions.js
+const asDate = (r) => {
+  let temp;
+  const dateSinceUnixEpoch = (days) => {
+    // Create a date at 00:00:00 1st Jan 1970 _in the current timezone_
+    const date = new Date(1970, 0, 1);
+    date.setDate(1 + days);
+    return date;
+  };
+  switch(r.t) {
+  case 'bool': return new Date(NaN);
+  case 'date': return r.v;
+  case 'num':  return dateSinceUnixEpoch(r.v);
+  case 'arr':
+  default:
+    r = asString(r);
+    if(RAW_NUMBER.test(r)) {
+      return dateSinceUnixEpoch(parseInt(r, 10));
+    } else if(DATE_STRING.test(r)) {
```

Comment:
Not sure if I am misunderstanding you here or if this code was confusing when you first looked at it (since the code was all weirdly formatted).  I think I still want this code in an `else if` block since is a different code flow to execute if my input matches the DATE_STRING than just jumping down to line 113 and just trying to return a Date with whatever input was provided.  

Let me know if there is still something you want changed here.

Reply:
https://eslint.org/docs/rules/no-else-return